librz/core/cmd/cmd_open.c

```diff
@@ -1251,3 +778,322 @@ RZ_IPI RzCmdStatus rz_open_binary_reload_handler(RzCore *core, int argc, const c
 	rz_core_block_read(core);
 	return RZ_CMD_STATUS_OK;
 }
+
+static RzCmdStatus open_file(RzCore *core, const char *filepath, ut64 addr, int perms, bool write_mode) {
+	RzCoreFile *cfile = rz_core_file_open(core, filepath, perms, addr);
+	if (!cfile) {
+		RZ_LOG_ERROR("Cannot open file '%s'\n", filepath);
+		return RZ_CMD_STATUS_ERROR;
+	}
+
+	core->num->value = cfile->fd;
+	if (addr == 0) { // if no baddr defined, use the one provided by the file
+		addr = UT64_MAX;
+	}
+	rz_core_bin_load(core, filepath, addr);
+	if (write_mode) {
+		RzIODesc *desc = rz_io_desc_get(core->io, cfile->fd);
+		if (!desc || !(desc->perm & RZ_PERM_W)) {
+			RZ_LOG_WARN("Cannot make maps for %s writable.\n", filepath);
+			return RZ_CMD_STATUS_ERROR;
+		}
+		void **it;
+		rz_pvector_foreach (&cfile->maps, it) {
+			RzIOMap *map = *it;
+			map->perm |= RZ_PERM_WX;
+		}
+	}
+
+	rz_core_block_read(core);
```

Comment:
Please handle the errors/check the return value

Reply:
this is never checked anywhere in the code base and I wouldn't know what to do in case of failure either. I don't think I should make the whole thing fail because of this.

Reply:
there is no reason to check that, because when it fails to read, the buffer is filled with 0xff as per design